### PR TITLE
Add support for DiagnosticSuppressors

### LINKED ISSUE #165: Support testing DiagnosticSuppressor

### LINKED ISSUE #253: Verify doesn't allow diagnostics with the same id
This is the same as #99 which was then solved by the writer to make all id's unique.

I ran into this again with another Analyzer.
Looking at the Roslyn analyzer they also use multiple DiagnosticsDescriptors with the same Id, just with different message/title.
See:

* [IdentifiersShouldNotMatchKeywords](https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotMatchKeywords.cs)
- [DefineAccessorsForAttributeArguments](https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DefineAccessorsForAttributeArguments.cs)

There are several others.

So I think the Unique test in RoslynAssert.cs `VerifyAnalyzerSupportsDiagnostic` should be removed.

Currently `ExpectedDiagnostic` only works on the RuleId, maybe it should work on `DiagnosticDescriptor` instead which then needs to be one listed in the `SupportedDiagnostics`.

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #165 

I opted for the same set of overloads as for `RoslynAssert.Diagnostics`, but are open for change there.

Priniciple is:
- Check the code for '↓' or use `ExpectedDiagnostics`
- Compile code without `DiagnoticSuppressor` to ensure that `ExpectedDiagnostic` is raised.
- Compile code with the `DiagnosticSuppressor` and check if the diagnostic `IsSuppressed`.

I tried it out with one of my analyzers and it seems to work fine.
Once this is in and release, I will update nunit.analyzers to use it as well, dropping the code there.

Initially I went for different approach, but the 2nd version aligns much better with the existing framework.
I have not looked at the analyzer to see what needs changing there so that it warns of missing '↓'. It doesn't do so at the moment.

Fixes #253 

Not related and we can split this into two PRs is you prefer.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -172,3 +172,15 @@ csharp_space_between_square_brackets = false
...
+csharp_using_directive_placement = inside_namespace:error
+
+# IDE0045: Convert to conditional expression
+dotnet_style_prefer_conditional_expression_over_assignment = false
+
+# IDE0046: Convert to conditional expression
+dotnet_style_prefer_conditional_expression_over_return = false
+
+# IDE0061: Use expression body for local functions
+csharp_style_expression_bodied_local_functions = false
```
Comment (JohanLarsson) [inline .editorconfig:186]: Not a problem to marge but stuff like this is nice in separate PR.
Just a minor nit not at all a problem.

**Comment by JohanLarsson** [comment]
Very nice, thanks!
I'll try to find time to release a new nuget today.

File: .editorconfig (hunk 1 above)
Comment (manfred-brands) [reply .editorconfig:186]: Main thing was to reduce reported issues before I started modifying things.

**Comment by JohanLarsson** [comment]
https://www.nuget.org/packages/Gu.Roslyn.Asserts/4.1.0

**Comment by manfred-brands** [comment]
Thanks

**Pull request merged**